### PR TITLE
Refactoring of statistics

### PR DESCRIPTION
**Pull request by musilto7**
I've read the code of computing statistics. There is int variable called mType which can have only three values. I decided to create enum called AxisType  and replace  type of mType.
There is method calculateDone in Stats.java. Second method variable is boolean. The function of boolean is not described in javadoc. I decided replace boolean by Chartype and make the method private. Then two public methods will be created: calculateReviewCount and calculateReviewTime. This two new methods will use method calculateDone.

### CODE REVIEW

**Comment by timrae** [comment]
Does this fix any bugs or anything? What's the reason for the changes?

**Comment by musilto7** [comment]
It is pure refactoring. The motivatation is make the code more readable.  It is my first attempt for commit to open source project, next time I will try bug fixing or implementing of the new feature :)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Stats.java [hunk 1]
```diff
@@ -365,17 +367,32 @@ private boolean calculateDue(int type) {
...
-        if (reps) {
+        if (charType == ChartType.REVIEW_COUNT) {
             mTitle = R.string.stats_review_count;
-            mAxisTitles = new int[] { type, R.string.stats_answers, R.string.stats_cumulative_answers };
-        } else {
-            mTitle = R.string.stats_review_time;
-        }
+            mAxisTitles = new int[] { type.ordinal(), R.string.stats_answers, R.string.stats_cumulative_answers };
+        } else if(charType == ChartType.REVIEW_TIME)
+            {
```
Comment (hssm) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Stats.java]: Make sure this opening brace `{` is on the same line as the beginning of the block to remain consistent with the rest of the codebase. Also, the indenting of this block is off.

**Comment by hssm** [comment]
Everything looks fine here. Just one minor nitpick for you to fix up. Once you've fixed it, you can amend the commit and force push the branch and it'll show up here.

> It is my first attempt for commit to open source project

Refactoring code is always a great way to familiarize yourself with the codebase. Don't feel compelled to rush into adding new features or anything if you're comfortable with doing just this for a while. Thanks for taking the time to help us improve AnkiDroid!

**Comment by musilto7** [comment]
I fixed it. But the tests failed. I tried to start the tests on my local machine and they passed. I am not sure what goes wrong.

**Comment by timrae** [comment]
Travis is going crazy lately, failing for no reason. Don't worry about it. Can you please squash your commits though:

https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#squashing-a-series-of-commits-down-to-one

**Comment by musilto7** [comment]
Squashing is done.

**Comment by hssm** [comment]
In the future, if you are only making a small change to an existing commit, you can simply amend the previous one instead of committing on top of it:
```git commit --amend```

This will save you from having to squash it down later on.

**Pull request merged**